peps/pep-0725.rst

```diff
@@ -217,15 +223,52 @@ There is no ready-made support for virtual packages in PURL or another
 standard. There are a relatively limited number of such dependencies though,
 and adopting a scheme similar to PURL but with the ``virtual:`` rather than
 ``pkg:`` scheme seems like it will be understandable and map well to Linux
-distros with virtual packages and to the likes of Conda and Spack.
+distros with virtual packages and to the likes of conda and Spack.
 
 The two known virtual package types are ``compiler`` and ``interface``.
 
+Note: Adding a new ``abstract/`` type to PURL has been
+`proposed <https://github.com/package-url/purl-spec/issues/222>`__ in the
+meantime. If that goes through, ``virtual:`` PURLs can be converted to
+``pkg:abstract/`` PURLs and understood as equivalent.
+
 Versioning
 ''''''''''
 
+PURLs support fixed versions via the ``@`` component of the URL. For example,
+``numpy===2.0`` can be expressed as ``pkg:pypi/numpy@2.0``.
+
 Support in PURL for version expressions and ranges beyond a fixed version is
-still pending, see the Open Issues section.
+available via the ``vers`` scheme. Users are supposed to couple a ``pkg:``
+URL with a ``vers`` URL. For example, to express ``numpy>=2.0``, the PURL
+equivalent would be ``pkg:pypi/numpy`` plus ``vers:pypi/>=2.0``. This
+can be expressed as a two-item list: ``["pkg:pypi/numpy",
+"vers:pypi/>=2.0"]``.
+
+However, this is not very ergonomic. We propose a PURL-compatible syntax that
+extends the ``@`` component to support version ranges directly:
+
+.. JRG: I suggested 'dep:' (dependency) but it could also be 'req:' (requirement)
+
+- The scheme for this PURL derivative is ``dep:`` (for dependency).
+- When no operator is present, the field is understood as a fixed version.
+- The version range syntax must be compatible with PEP 440. As a result, it is
+  equivalent to ``vers:pypi/`` PURLs.
```

Comment:
The problem with this is that it won't be able to express some versions, particularly OpenSSL versions (that use letters `a`, `b`, … as post-releases, rather than prereleases.

Reply:
Yep, same with libjpeg IIRC. In those cases I think we need to fallback to `[pkg, vers]` tuples. Thankfully versions like `1.1.1a` are illegal in PEP440, so at least there's no ambiguity there.

Reply:
I think I'm fine with declaring that "out of scope - corner case that can't be expressed" if we don't want the extra complexity of really allowing `[pkg, vers]` tuples. Can still mostly express it with `>1.1.1,<1.1.2.dev0` anyway.